service/updaterconf/updaterconf.go

```diff
@@ -0,0 +1,293 @@
+	u.versionFilename = versionFileName
+	return u
+}
+
+// CurrentVersion reads the current update version from file
+func (u *UpdaterConf) CurrentVersion() string {
+	return osutils.ReadStringFromFile(u.getFilePath(u.versionFilename), "")
+}
+
+// IsReloading returns if the reloading flag been set to instruct service reload in progress.
+func (u *UpdaterConf) IsReloading() bool {
+	reloadFilePath := u.getReloadFilePath()
+	return osutils.FileExists(reloadFilePath)
+}
+
+func (u *UpdaterConf) getFilePath(filename string) string {
+	return filepath.Join(u.silverDir, filename)
+}
+
+func (u *UpdaterConf) getServiceConfigPath() string {
+	return u.getFilePath(u.silverConfigFilename)
+}
+
+func (u *UpdaterConf) getBackupServiceConfigPath() string {
+	return u.getFilePath(u.backupFilePrefix + u.silverConfigFilename)
+}
+
+func (u *UpdaterConf) getReloadFilePath() string {
+	// No support for custom reload files in updater yet
+	return u.getFilePath(config.ReloadFileName)
+}
+
+// EnableAutoUpdates enables auto updates
+func (u *UpdaterConf) EnableAutoUpdates() error {
+	u.lock.Lock()
+	defer u.lock.Unlock()
+
+	reloading := u.IsReloading()
+	if reloading {
+		return fmt.Errorf("app is already reloading, cannot enable auto updates until reloading complete")
+	}
+	enabled, err := u.IsAutoUpdateEnabled()
+	if err != nil {
+		return fmt.Errorf("unable to determine if updates already enabled: %w", err)
+	} else if enabled {
+		fmt.Printf("auto update already enabled\n")
+		return nil
+	}
+
+	fmt.Printf("enabling auto updates\n")
+	err = osutils.CopyFile(u.getBackupServiceConfigPath(), u.getServiceConfigPath())
+	if err != nil {
+		fmt.Printf("enableUpdates could not copy file: %v", err)
+		return err
+	}
+	err = u.reloadApp()
+	if err != nil {
+		fmt.Printf("enableUpdates reloadApp error: %v\n", err)
+		return err
+	}
+	fmt.Printf("auto updates enabled\n")
+	return nil
+}
+
+// DisableAutoUpdate disables auto updates, backing up the active configuration file
+func (u *UpdaterConf) DisableAutoUpdate() error {
+	u.lock.Lock()
+	defer u.lock.Unlock()
+
+	reloading := u.IsReloading()
+	if reloading {
+		return fmt.Errorf("app is already reloading, cannot disable auto updates until reloading complete")
+	}
+	enabled, err := u.IsAutoUpdateEnabled()
+	if err != nil {
+		return fmt.Errorf("unable to determine if updates already enabled: %w", err)
+	} else if !enabled {
+		fmt.Printf("auto update already disabled\n")
+		return nil
+	}
+	fmt.Printf("disabling auto updates\n")
+	err = u.backupServiceConfigIfRequired()
+	if err != nil {
+		fmt.Printf("disableUpdates could not create default update config: %v\n", err)
+		return err
+	}
+	conf, err := u.loadConfig(u.getServiceConfigPath())
+	if err != nil {
+		fmt.Printf("disableUpdates could not load config file: %v\n", err)
+		return err
+	}
+
+	_, ok := conf[startupTasksKey]
+	if ok {
+		delete(conf, startupTasksKey)
+	}
+	_, ok = conf[scheduledTasksKey]
```

Comment:
This will remove all tasks, but is there a chance there are tasks which is not the updater?

Reply:
Yep you're right, whilst its unlikely this to be the case, in the context of silver this is a general purpose feature that needs to be aware of this distinction

Reply:
updated with all the acrobatics required to deal with filtering at the map level 😩